src/org/javarosa/form/api/FormEntryController.java

```diff
@@ -72,12 +72,13 @@ public int answerQuestion(IAnswerData data, boolean midSurvey) {
 
 
     /**
-     * Attempts to save the answer at the specified FormIndex into the
-     * datamodel.
+     * Attempts to save the answer at the specified FormIndex into the datamodel.
      *
-     * @param index
-     * @param data
-     * @return OK if save was successful, error if a constraint was violated.
+     * Side effects: Mutates `index` by setting its value property to the value of `data`.
```

Comment:
This is not quite right -- `index` is not mutated. You could say that it sets the answer on the TreeElement corresponding to the provided index, perhaps.

Reply:
Updated! Thanks for pointing out both of those corrections.